src/Microsoft.ML.Data/Transforms/DropSlotsTransform.cs

```diff
@@ -503,22 +508,22 @@ private static bool IsValidColumnType(ColumnType type)
 
                 categoricalRanges = null;
                 var typeSrc = _srcTypes[iinfo];
-                if (!typeSrc.IsVector)
+                if (!(typeSrc is VectorType vectorType))
```

Comment:
Not necessary to change, but whenever this pattern happens it always confuses me... `if (!(a is B b) {} else {... usage of b ...}`. 😄

Reply:
I agree I don't like the pattern either... but I didn't want to change the logic too much. And in the situations where you want to `return` or `throw` when the object isn't a type, this is probably the best way to do it. So it felt useless to try to avoid it.